### ISSUE TITLE
Deleting Edge layer from a geopackage makes it impossible to open in QGIS anymore

### ISSUE THREAD

**Issue by deltamarnix** [Deltares/Ribasim#1684]
- Open QGIS and ensure that the Ribasim plugin is installed and enabled.
- Open the application via the Ribasim button on the QGIS toolbar: _Ribasim panel opens_.
- Press the "New" button in the Model tab: _file navigation window pops up_.
- Fill in a name ("test1").
- Press OK: _A toml and database file are created on the given location_.
- Press the "Edge" layer in the Model tab, press "Remove from Dataset": _Popup shows "Deleting: Edge"_.
- Press "Yes": "Edge" layer is removed from Layers panel and from Model tab.
- Press the "Open" button in the Model tab: _file navigation window pops up_.
- Open "test1.toml": _Throws error, because it cannot find "Edge anymore"_.

Expected: Not such a big crash, but more like a validation error. Or we should not allow deletion of "Edge" and "Node" layers from the database.

**Comment by Huite** [comment]
Indeed, it shouldn't be possible to delete the Node and Edge layers.

In QGIS-Tim, I have two mandatory layers as well, they are always removed from the deletion selection:

```python
        # Collect the selected items
        selection = self.selectedItems()
        selection = [
            item
            for item in selection
            if not isinstance(item.element, (Aquifer, Domain))
        ]

```

This is currently silent though, you probably want a an info message that mentions that these layers cannot be deleted.
